Tests/RxSwiftTests/Tests/Observable+MultipleTest.swift

```diff
@@ -4621,4 +4621,85 @@ extension ObservableMultipleTest {
             Subscription(200, 400)
             ])
     }
-}
\ No newline at end of file
+}
+
+// MARK: forwardIf
+extension ObservableMultipleTest {
+    func testForwardIf_simple1() {
+        let scheduler = TestScheduler(initialClock: 0)
+        
+        let xs = scheduler.createHotObservable([
+            next(90, 1),
+            next(180, 2),
+            next(250, 3),
+            next(260, 4),
+            next(310, 5),
+            next(360, 6),
+            completed(390)
+        ])
+        
+        let ys = scheduler.createHotObservable([
+            next(210, true),
+            next(300, false),
+            next(350, true),
+            completed(400)
+        ])
+
+        let res = scheduler.start {
+            xs.forwardIf(ys)
+        }
+
+        XCTAssertEqual(res.events, [
+            next(250, 3),
+            next(260, 4),
+            next(360, 6),
```

Comment:
I think if you want to use your operator in these contexts
```
service.refreshModel
  .forwardIf(visible)
  .subscribeNext(bindModel)
  .addDisposableTo(disposeBag)
```
.. then IMHO this should be 
```
XCTAssertEqual(res.events, [
 +            next(250, 3),
 +            next(260, 4),
 +            next(350, 5), // <-- replaying last element received
 +            next(360, 6),
```